lib/sshutils/ctx.go

```diff
@@ -29,6 +32,7 @@ import (
 
 // ConnectionContext manages connection-level state.
 type ConnectionContext struct {
+	context.Context
```

Comment:
Don't embed the context, pass it via func arguments instead: https://github.com/golang/go/wiki/CodeReviewComments#contexts
For premature cancellation, the caller can wrap using `context.WithCancel`.
For regular cleanup, consider adding a `Close` method where necessary

Reply:
I've removed the other cases of contexts in structs from this PR, but this (and `srv.ServerContext`) is a bit different.  My thinking about embedding in these cases is that these already were basically `Context` values that just happened to not implement the full interface.  They already where literally request-scoped parameters which carry some request-level state and also assist in signalling cancellation (they even have a `Context`-esque parent-child relationship).  I decided to embed `context.Context` here because it was simpler and easier to just make it official.